include/dgl/runtime/ndarray.h

```diff
@@ -415,6 +415,9 @@ struct NDArray::Container {
   std::vector<int64_t> stride_;
   /*! \brief The internal array object */
   std::atomic<int> ref_counter_{0};
+
+  bool pinned_by_dgl_{false};
```

Comment:
suggest to just call it `is_pinned_`

Reply:
If we borrowed a tensor pinned by PyTorch, we don't want to unpin it automatically.